src/core/lib/security/credentials/ssl/ssl_credentials.h

```diff
@@ -70,6 +70,11 @@ class grpc_ssl_credentials : public grpc_channel_credentials {
                     const grpc_ssl_verify_peer_options* verify_options);
 
   grpc_ssl_config config_;
+  grpc_security_status initialize_client_handshaker_factory(
+      const grpc_ssl_config* config, const char* pem_root_certs,
+      const tsi_ssl_root_certs_store* root_store,
+      tsi_ssl_session_cache* ssl_session_cache);
+  tsi_ssl_client_handshaker_factory* client_handshaker_factory_ = nullptr;
```

Comment:
Can we store this as a grpc_core::RefCounterPtr instead?

This also means that we will have to pass it around as a grpc_core::RefCounterPtr instead of as a raw pointer.

Rationale: We have to be very careful with lifetimes here, in case the user destroys the credentials object once the channel is created but before the TLS handshake is complete.

Reply:
That makes a lot of sense, will make the change

Reply:
I realized the handshaker_factories are already doing ref counting c-style, added explicit code to make this clear.

See [here](https://github.com/grpc/grpc/blob/5c9e4050c1e8c66b1b8b4e93d48b8e23ff1ea818/src/core/tsi/ssl_transport_security.cc#L1759) for an example of where it is already being counted.